code/game/machinery/seed_extractor.dm

```diff
@@ -74,3 +106,102 @@ obj/machinery/seed_extractor/attackby(var/obj/item/O as obj, var/mob/user as mob
 	..()
 
 	return
+//Code shamelessly ported over from tgstation's github repo, PR #2973, credit to Kelenius for the original code
+datum/seed_pile
+	var/name = ""
+	var/lifespan = 0	//Saved stats
+	var/endurance = 0
+	var/maturation = 0
+	var/production = 0
+	var/yield = 0
+	var/potency = 0
+	var/amount = 0
+
+datum/seed_pile/New(var/name, var/life, var/endur, var/matur, var/prod, var/yie, var/poten, var/am = 1)
+	src.name = name
+	src.lifespan = life
+	src.endurance = endur
+	src.maturation = matur
+	src.production = prod
+	src.yield = yie
+	src.potency = poten
+	src.amount = am
+
+/obj/machinery/seed_extractor/attack_hand(mob/user as mob)
+	user.set_machine(src)

```

Comment:
I couldn't really understand what this did even after checking with the IRC, I just moved it to interact like you said. If it's still wrong please do tell.